src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/CodeGeneration/CodeGenerationOptions.cs

```diff
@@ -38,6 +37,7 @@ private protected CodeGenerationOptions()
     private protected CodeGenerationOptions(IOptionsReader options, CodeGenerationOptions fallbackOptions, string language)
     {
         NamingStyle = options.GetOption(NamingStyleOptions.NamingPreferences, language, fallbackOptions.NamingStyle);
+        AllowGenerateInHiddenCode = fallbackOptions.AllowGenerateInHiddenCode;
```

Comment:
I'd add a comment that this option is not read from `options` since it isn't persistable. Instead, the code working with the `CodeGenerationOptions` sets it directly.